tests/e2e/feature-validation.py

```diff
@@ -0,0 +1,99 @@
+import pandas as pd
+import pytest
+import pytz
+import uuid
+import time
+from datetime import datetime
+
+from feast.client import Client
+from feast.entity import Entity
+from feast.feature import Feature
+from feast.feature_set import FeatureSet
+from feast.type_map import ValueType
+from google.protobuf import json_format
+from google.protobuf.duration_pb2 import Duration
+from tensorflow_metadata.proto.v0 import statistics_pb2
+
+pd.set_option("display.max_columns", None)
+
+PROJECT_NAME = "batch_" + uuid.uuid4().hex.upper()[0:6]
+
+
+@pytest.fixture(scope="module")
+def client(core_url, allow_dirty):
+    # Get client for core and serving
+    client = Client(core_url=core_url)
+    client.create_project(PROJECT_NAME)
+    client.set_project(PROJECT_NAME)
+
+    # Ensure Feast core is active, but empty
+    if not allow_dirty:
+        feature_sets = client.list_feature_sets()
+        if len(feature_sets) > 0:
+            raise Exception(
+                "Feast cannot have existing feature sets registered. Exiting tests."
+            )
+
+    return client
+
+
+@pytest.fixture(scope="module")
+def dataset_basic(client):
+    fv_fs = FeatureSet(
+        "feature_validation",
+        features=[
+            Feature("strings", ValueType.STRING),
+            Feature("ints", ValueType.INT64),
+            Feature("floats", ValueType.FLOAT),
+        ],
+        entities=[Entity("entity_id", ValueType.INT64)],
+        max_age=Duration(seconds=100),
+    )
+    client.apply(fv_fs)
+    time.sleep(20)
+
+    N_ROWS = 20
+
+    time_offset = datetime.utcnow().replace(tzinfo=pytz.utc)
+    df = pd.DataFrame(
+        {
+            "datetime": [time_offset] * N_ROWS,
+            "entity_id": [i for i in range(N_ROWS)],
+            "strings": ["a", "b"] * int(N_ROWS / 2),
+            "ints": [int(i) for i in range(N_ROWS)],
+            "floats": [10.5 - i for i in range(N_ROWS)],
+        }
+    )
+
+    expected_stats = statistics_pb2.DatasetFeatureStatisticsList()
+    with open("statistics/expected_output_basic_dataset.json", "r") as fo:
```

Comment:
I think it would be less fragile if we generated the statistics from the dataframe using TFDV instead of reading it from a file. That would make it a lot easier to ensure that we have and can maintain compatibility.

Reply:
We don't support the full set of TFDV stats, so generating it on the fly wouldn't allow the test to pass.

Reply:
Possibly to filter out the ones we don't need?